drivers/include/touch_dev.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_touch_dev Touch device generic API
+ * @ingroup     drivers_misc
+ * @brief       Define the generic API of a touch device
+ * @experimental This API is experimental and in an early state - expect
+ *               changes!
+ * @{
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef TOUCH_DEV_H
+#define TOUCH_DEV_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * @brief   Forward declaration for touch device struct
+ */
+typedef struct touch_dev touch_dev_t;
+
+/**
+ * @brief   Touch position coordinates
+ */
+typedef struct {
+    uint16_t x;     /**< X coordinate */
+    uint16_t y;     /**< X coordinate */
+} touch_position_t;
+
+/**
+ * @brief   Generic type for a touch driver
+ */
+typedef struct {
+
+    /**
+     * @brief   Get the height of the touch device
+     *
+     * @param[in] dev       Pointer to the touch device
+     *
+     * @return              Height in points
+     */
+    uint16_t (*height)(touch_dev_t *dev);
+
+    /**
+     * @brief   Get the width of the touch device
+     *
+     * @param[in] dev       Pointer to the touch device
+     *
+     * @return              Width in points
+     */
+    uint16_t (*width)(touch_dev_t *dev);
+
+    /**
+     * @brief   Get the current pressed position of the touch device
+     *
+     * @param[in] dev       Pointer to the touch device
+     * @param[out] position The position pressed
```

Comment:
What about also returning `bool` to indicate if it is being pressed or not and the position in one function call?

Reply:
I was looking at #13214 and maybe we should return the number of touch points and 0 means no touch. We could indeed remove the is_pressed function from the API.
What do you think ?

Reply:
Would then make sense to pass an array of `touch_position_t` here? meaning that we should also indicate the length of it. I would remove `is_pressed` if we allow here to pass `NULL` pointer to just ask for the number of touches.

Reply:
Ok, let's do this then.